ddtrace/llmobs/_integrations/utils.py

```diff
@@ -397,6 +397,209 @@ def openai_set_meta_tags_from_chat(span: Span, kwargs: Dict[str, Any], messages:
+                    action = json.loads(f'{{"action": {action_json_str}}}')
+
+            # function call
+            # Use output.id as the tool_id
+            tool_id = _get_attr(output, "id", "")
+            tool_name = _get_attr(output, "name", "")
+            arguments = _get_attr(output, "arguments", "")
+            if _get_attr(output, "arguments", ""):
+                arguments = json.loads(arguments)
+            # Set reasoning action as the parameter
+            elif _get_attr(output, "action", ""):
+                arguments = action
+            # Set file search queries as the parameter
+            elif _get_attr(output, "queries", None):
+                arguments = file_queries
+
+            tool_call = {
+                "name": tool_name,
+                "tool_id": tool_id,
+                "type": output_type,
+            }
+            if arguments:
+                tool_call["arguments"] = arguments
+            tool_call_info.append(tool_call)
+            arguments = tool_call.get("arguments", {})
+            # Convert arguments to string before dispatching
+            arguments_str = json.dumps(arguments) if isinstance(arguments, (dict, list)) else str(arguments)
+            core.dispatch(
+                DISPATCH_ON_LLM_TOOL_CHOICE,
+                (
+                    tool_id,
+                    tool_name,
+                    arguments_str,
+                    {
+                        "trace_id": format_trace_id(span.trace_id),
+                        "span_id": str(span.span_id),
+                    },
+                ),
+            )
+            if tool_call_info:
+                output_messages.append({"content": "", "tool_calls": tool_call_info})
+                core.dispatch(DISPATCH_ON_TOOL_CALL_OUTPUT_USED, (tool_id, span))
+        span._set_ctx_item(OUTPUT_MESSAGES, output_messages)
+
+
+def openai_construct_response_from_streamed_chunks(streamed_chunks: List[Any]) -> Dict[str, Any]:
+    """Constructs a response dictionary from streamed chunks.
+    The resulting response dictionary is of form:
+    {"output": [{"role": "...", "content": [{"text": "..."}]}]}
+    """
+    message: Dict[str, Any] = {"content": "", "tool_calls": []}
+
+    for chunk in streamed_chunks:
+        if not isinstance(chunk, tuple) or len(chunk) != 2:
+            continue
+
+        chunk_type, chunk_value = chunk
+        # print(f"Processing chunk type: {chunk_type}, value: {chunk_value}")
+        if chunk_type == "usage":
+            """
+            TODO: currently chunk_value is a ResponseUsage object.
+            We want to convert it to a dict that looks like 
+            {"input_tokens": ..., "output_tokens": ..., "total_tokens": ...}
+            """
+            message["usage"] = chunk_value
+
+        if chunk_type == "output" and isinstance(chunk_value, list):
+            for output in chunk_value:
+                if hasattr(output, "content"):
```

Comment:
Might be worth adding a comment on each if/elif case to denote what output type we're trying to extract